EmaPajic/assignment3/assignment3.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+@author: EmaPajic
+"""
+
+from dictionary import Dictionary
+
+def out_of_bounds(grid,i,j,n,m,visited):
+    #checks if coordinates are out of bounds or if it is already visited
+    if i < 0 or j < 0:
+        return True
+    if i >= n or j>= m:
+        return True
+    if visited[tuple((i,j))] == 1:
+        return True
+    return False
+
+def find_words_from_grid(grid,i,j,n,m,word,dictionary,foundWords,visited):
+    #function that finds all words from grid that are in dictionary and returns them in set, i and j are starting positions of search, word is our current word found
+    if out_of_bounds(grid,i,j,n,m,visited):
+        return
+    word += grid[i][j]
+    visited[tuple((i,j))] = 1
+    if dictionary.is_word(word):
+        foundWords.add(word)
+    elif dictionary.is_prefix(word):
+        find_words_from_grid(grid,i,j+1,n,m,word,dictionary,foundWords,visited)
```

Comment:
Can you generate the indexes with for loops?

Reply:
updated :)